onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -1145,6 +1145,23 @@ ORT_API_STATUS_IMPL(OrtApis::FillStringTensorElement, _Inout_ OrtValue* value, _
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtApis::GetStringTensorElementBuffer, _Inout_ OrtValue* value,
+                    _In_ size_t index, _In_ size_t length_in_bytes, _Inout_ char** buffer) {
+  TENSOR_READWRITE_API_BEGIN
+  auto* dst = tensor->MutableData<std::string>();
+  const auto len = static_cast<size_t>(tensor->Shape().Size());
+
+  if (index >= len) {
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "element index is out of bounds");
+  }
+
+  auto& s = dst[index];
+  s.resize(length_in_bytes);
```

Comment:
As this resize could truncate existing data or grow the buffer, maybe 'GetResizedStringTensorElementBuffer' would clarify the behavior as the current name make it sound like a simple getter.